config.c

```diff
@@ -85,7 +154,10 @@ int parse_config_arguments(struct mako_config *config, int argc, char **argv) {
 		} else if (strcmp(name, "height") == 0) {
 			config->height = strtol(optarg, NULL, 10);
 		} else if (strcmp(name, "margin") == 0) {
-			config->margin = strtol(optarg, NULL, 10);
+			if (parse_directional(optarg, &config->margin)) {
+				printf("Unable to parse margins\n\n");
```

Comment:
Nitpick: only one `\n`, and use `fprintf(stderr, …);` to print ot the error output

Reply:
Done!